Server/Controllers/WalletController.cs

```diff
@@ -26,15 +29,44 @@ public WalletController(ApplicationDbContext context,UserManager<ApplicationUser
         public List<Wallet> GetWallets()
         {
             var userId = userManager.GetUserId(User);
-            var wallets = context.Users.Include(e=>e.Wallets).FirstOrDefault(e => e.Id == userId).Wallets;
+            var wallets = context.Users.Include(e => e.Wallets).FirstOrDefault(e => e.Id == userId).Wallets;
             return wallets;
 
         }
 
+        [HttpGet]
+        [Route("{id}")]
+        public Wallet GetWallet(Guid id)
+        {
+            var userId = userManager.GetUserId(User);
+            var wallet = context.Users.Include(x => x.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.FirstOrDefault(x => x.Id == id);
+            return wallet;
+        }
+
+        [HttpGet]
+        [Route("transactions")]
+        public List<Transaction> GetTransactions()
+        {
+            var transactions_list = new List<Transaction>();
+            var userId = userManager.GetUserId(User);
+            var id_list = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.ToList();
+            var transactions = context.Transactions.ToList();
```

Comment:
same as with users. You do not want to load everything in memory. Use linq to filter. It will filter in the db.

Reply:
var id_list = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.ToList();

            foreach(var id in id_list)
            {
                foreach(var t in context.Transactions)
                {
                    if (id.Id == t.SourceWalletId || id.Id == t.DestinationWalletId)
                        transactions_list.Add(t);
                }
            }

Reply:
var transactions_list = new List<Transaction>();
            var userId = userManager.GetUserId(User);
            var id_list = context.Users.Include(e => e.Wallets).FirstOrDefault(x => x.Id == userId).Wallets.ToList();

            foreach(var id in id_list)
            {
                 var t_list = context.Transactions.Where(e => e.SourceWalletId == id.Id || e.DestinationWalletId == id.Id).ToList();
                transactions_list.AddRange(t_list);
            }